packages/suite/src/actions/suite/metadataLabelingActions.ts

```diff
@@ -25,21 +32,44 @@ import type { MetadataAction } from './metadataActions';
 import * as metadataActions from './metadataActions';
 import * as metadataProviderActions from './metadataProviderActions';
 
+type LabelableEntity = ReturnType<ReturnType<typeof getLabelableEntities>>[number];
+
 export const getLabelableEntities =
     (deviceState: string) => (_dispatch: Dispatch, getState: GetState) =>
         selectLabelableEntities(getState(), deviceState);
 
-type LabelableEntity = ReturnType<ReturnType<typeof getLabelableEntities>>[number];
+export const getLabelableEntitiesDescriptors = () => (dispatch: Dispatch, getState: GetState) => {
+    const device = selectDevice(getState());
+
+    if (!device?.state) return [];
+
+    const entitites = dispatch(getLabelableEntities(device.state));
+
+    return entitites
+        .map(entity => {
+            if ('key' in entity) return entity.key;
+            if ('state' in entity && entity.state) return entity.state;
+            throw new Error('entity without unique identifier');
```

Comment:
Can we use `reduce` here and filter the invalid entities out? This would (probably) crash the app.